### PR TITLE
refactor: maked app`a refactoring

### PR DESCRIPTION
**Pull request by Ilya-m2195**

### CODE REVIEW

File: src/App/store/store.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import { configureStore } from '@reduxjs/toolkit';
+import userSliceReducer from './userSlice';
+
+export const store = configureStore({
+  reducer: {
+    userSlice: userSliceReducer
+  }
+});
+
```
Comment (suveiko) [inline src/App/store/store.ts:9]: `export const useAppSelector: TypedUseSelectorHook<RootState> = useSelector`

add this also

File: src/App/store/userSlice.ts [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+export const postUser = createAsyncThunk<Promise<void>, IUserForm>(
+  'userSlice/postUser',
+  async (userInfo) => {
+    await fetch(URL, {
+      method: 'POST',
+      headers: { 'Content-Type': 'application/json' },
+      body: JSON.stringify(userInfo)
+    });
+  }
+);
```
Comment (suveiko) [inline src/App/store/userSlice.ts]: move this to separate function

File: src/components/AdvantagesInputGroup/AdvantagesInputGroup.tsx [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  ));
+
+  return (
+    <Box maw={300}>
+      <h3>Advantages</h3>
+      {fields}
+      <Group justify='start' mt='md'>
+        <Button
+          onClick={() => form.insertListItem('advantages', { advantage: '', key: randomId() })}
+        >
```
Comment (suveiko) [inline src/components/AdvantagesInputGroup/AdvantagesInputGroup.tsx:29]: move to function

File: src/components/StartingForm/StartingForm.tsx [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+      <Group justify='flex-start'>
+        <Button
+          type='button'
+          onClick={() => {
+            if (form.validateField('phone').hasError || form.validateField('email').hasError) {
+              return;
+            }
+            setIsStartFilling(true);
+          }}
+        >
```
Comment (suveiko) [inline src/components/StartingForm/StartingForm.tsx]: move this logic outside, create separate function for this

File: src/pages/StartingPage/StartingPage.tsx [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+        /^\w+([.-]?\w+)*@\w+([.-]?\w+)*(\.\w{2,})+$/.test(value) ? null : 'Invalid email',
+      name: (value) => (value.length > 50 ? 'Name must contain no more than 50 letters' : null),
+      surname: (value) => (value.length > 50 ? 'Name must contain no more than 50 letters' : null),
+      nickname: (value) => (value.length > 30 ? 'Name must contain no more than 30 letters' : null),
+      about: (value) =>
+        value.split(' ').join('').length > 200
+          ? 'about must contain no more than 200 letters'
+          : null
+    }
+  });
```
Comment (suveiko) [inline src/pages/StartingPage/StartingPage.tsx]: move this logic to helper

File: src/pages/StepPagesWrapper/StepPagesWrapper.tsx [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+  p: '110px 62px'
+};
+
+type Props = {
+  form: TypeForm;
+  setIsStartFilling: (value: boolean) => void;
+};
+
+export const StepPagesWrapper: FC<Props> = ({ form, setIsStartFilling }) => {
+  const value = useSelector((state: RootState) => state.userSlice.loading);
```
Comment (suveiko) [inline src/pages/StepPagesWrapper/StepPagesWrapper.tsx]: use useAppSelector for this

File: src/components/AvatarCustom/AvatarCustom.tsx [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+    return;
+  }
+  return name.split('')[0].toUpperCase();
+};
+
+export const AvatarCustom: FC<Props> = ({ name, surname }) => {
+  return (
+    <div className={styles.container}>
+      <span className={styles.text}>{firstLitter(name!) || 'И'}</span>
+      <span className={styles.text}>{firstLitter(surname!) || 'И'}</span>
```
Comment (suveiko) [inline src/components/AvatarCustom/AvatarCustom.tsx]: you can do this by css property capitalize

File: src/pages/StartingPage/StartingPage.tsx [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+  return (
+    <div>
+      <Header name={form.values.name} surname={form.values.surname} />
+      <form onSubmit={form.onSubmit((values: IUserForm) => dispatch(postUser(values)))}>
```
Comment (suveiko) [inline src/pages/StartingPage/StartingPage.tsx]: move onSubmit logic above

File: src/components/AdvantagesInputGroup/AdvantagesInputGroup.tsx [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+import { FC } from 'react';
+import { FaTrash } from 'react-icons/fa';
+
+export const AdvantagesInputGroup: FC<FormProps> = ({ form }) => {
+  const fields = form.values.advantages.map((item, index) => (
+    <Group key={item.key} mt='xs'>
+      <TextInput
+        placeholder='Enter your advantage'
+        withAsterisk
+        style={{ flex: 1 }}
```
Comment (AdammmVV) [inline src/components/AdvantagesInputGroup/AdvantagesInputGroup.tsx]: don't use inline styles

File: src/components/AdvantagesInputGroup/AdvantagesInputGroup.tsx [hunk 10]
```diff
@@ -0,0 +1,35 @@
...
+        {...form.getInputProps(`advantages.${index}.advantage`)}
+      />
+      <ActionIcon color='red' onClick={() => form.removeListItem('advantages', index)}>
```
Comment (AdammmVV) [inline src/components/AdvantagesInputGroup/AdvantagesInputGroup.tsx]: also put it into a function

File: src/components/AvatarCustom/AvatarCustom.tsx [hunk 11]
```diff
@@ -1,22 +1,23 @@
...
 import styles from './avatarCustom.module.css';
 
+type Props = {
+  name: string | undefined;
+  surname: string | undefined;
+};
+
 const firstLitter = (name: string): string | void => {
-  if (typeof name === 'undefined') {
+  if (name.length === 0) {
```
Comment (AdammmVV) [inline src/components/AvatarCustom/AvatarCustom.tsx]: just try "name"

File: src/components/AvatarCustom/AvatarCustom.tsx [hunk 12]
```diff
@@ -1,22 +1,23 @@
...
     return;
   }
   return name.split('')[0].toUpperCase();
 };
```
Comment (AdammmVV) [inline src/components/AvatarCustom/AvatarCustom.tsx]: take out to helpers

File: src/components/Header/Header.tsx [hunk 13]
```diff
@@ -3,13 +3,18 @@ import styles from './header.module.css';
...
+
+export const Header: FC<Props> = ({ name, surname }) => {
   return (
     <header>
       <div className={styles.header}>
-        <AvatarCustom />
+        <AvatarCustom name={name} surname={surname} />
         <div>
-          <h2>Иван Иванов</h2>
+          <h2>{name && surname ? `${name} ${surname}` : 'Иван Иванов'}</h2>
```
Comment (AdammmVV) [inline src/components/Header/Header.tsx]: use a variable

File: src/components/Modal/ModalError/ModalError.tsx [hunk 14]
```diff
@@ -0,0 +1,25 @@
...
+        <img src={errorIcon} alt='x' />
+      </Flex>
+      <Group justify='flex-end' mt='md'>
+        <Button variant='filled' size='md' onClick={onClose}>
+          Закрыть
+        </Button>
+      </Group>
+    </Paper>
+  );
+};
```
Comment (AdammmVV) [inline src/components/Modal/ModalError/ModalError.tsx:25]: try using [https://mantine.dev/theming/mantine-provider/](url)

File: src/pages/StepPagesWrapper/StepPagesWrapper.tsx [hunk 15]
```diff
@@ -0,0 +1,60 @@
...
+  const [active, setActive] = useState(0);
+  const returnStartPage = () => setIsStartFilling(false);
+  const nextStep = () => setActive((current) => (current < 3 ? current + 1 : current));
```
Comment (AdammmVV) [inline src/pages/StepPagesWrapper/StepPagesWrapper.tsx]: don't use magic numbers, try Enum

File: src/pages/StepPagesWrapper/StepPagesWrapper.tsx [hunk 16]
```diff
@@ -0,0 +1,60 @@
...
+          {value && <ModalWindow value={value} returnStartPage={returnStartPage} />}
+        </Stepper.Completed>
+      </Stepper>
+      {returnStep(active)}
+      <Group justify='space-between' mt='64'>
+        <Button variant='outline' onClick={active ? prevStep : returnStartPage}>
+          Назад
+        </Button>
+        <Button type={active <= 2 ? 'button' : 'submit'} onClick={nextStep}>
+          {active < 2 ? 'Далее' : 'Отправить'}
```
Comment (AdammmVV) [inline src/pages/StepPagesWrapper/StepPagesWrapper.tsx]: use variable

File: src/pages/StepPagesWrapper/StepPagesWrapper.tsx [hunk 17]
```diff
@@ -0,0 +1,60 @@
...
+        <Stepper.Completed>
+          {value && <ModalWindow value={value} returnStartPage={returnStartPage} />}
+        </Stepper.Completed>
+      </Stepper>
+      {returnStep(active)}
+      <Group justify='space-between' mt='64'>
+        <Button variant='outline' onClick={active ? prevStep : returnStartPage}>
+          Назад
+        </Button>
+        <Button type={active <= 2 ? 'button' : 'submit'} onClick={nextStep}>
```
Comment (AdammmVV) [inline src/pages/StepPagesWrapper/StepPagesWrapper.tsx]: use variable

File: src/components/Modal/ModalWindow.tsx [hunk 18]
```diff
@@ -0,0 +1,26 @@
...
+};
+
+export const ModalWindow: FC<Props> = ({ value, returnStartPage }) => {
+  const [isClose, setOnClose] = useState(true);
+  const onClose = () => setOnClose(false);
+
+  return (
+    <>
+      <Modal opened={isClose} centered onClose={onClose}>
+        {value === 'failed' ? (
```
Comment (AdammmVV) [inline src/components/Modal/ModalWindow.tsx]: use variable

File: src/App/store/userSlice.ts [hunk 19]
```diff
@@ -0,0 +1,38 @@
...
+  reducers: {},
+  extraReducers: (builder) => {
+    builder.addCase(postUser.pending, (state) => {
+      state.loading = 'pending';
+    });
+    builder.addCase(postUser.fulfilled, (state) => {
+      state.loading = 'succeeded';
+    });
+    builder.addCase(postUser.rejected, (state) => {
+      state.loading = 'failed';
```
Comment (AdammmVV) [inline src/App/store/userSlice.ts]: use Enum

**Pull request merged**